src/common/helpers/cache.ts

```diff
@@ -0,0 +1,30 @@
+import {get as _get} from 'lodash';
+import config from './config';
+import {createClient} from 'redis';
+
+
+// Set up Redis, which is used for sessions and to cache requests to external APIs
+const redisHost = _get(config, 'session.redis.host', 'localhost');
+const redisPort = _get(config, 'session.redis.port', 6379);
+const redisClient = createClient({
+	url: `redis://${redisHost}:${redisPort}`
+});
+
+export async function getCachedJSON<T>(cacheKey: string) {
+	if (redisClient.isReady) {
+		const cachedValue = await redisClient.get(cacheKey);
+		if (cachedValue) {
+			return JSON.parse(cachedValue) as T;
```

Comment:
I always wrap `JSON.parse` calls in a try…catch block, I've been bitten by malformed JSON too many times !

Reply:
While I usually let errors propagate until they reach the middleware layer, you're right that it makes more sense to handle them directly in the case of the cache.
In 0421c23 I will log an error and then fail deliberately, i.e. treat this case as a cache miss.